sys/kern/sysv_shm.c

```diff
@@ -1089,8 +1089,10 @@ static struct syscall_helper_data shm64_syscalls[] = {
 	FREEBSD64_SYSCALL_INIT_HELPER(freebsd64_shmat),
 	FREEBSD64_SYSCALL_INIT_HELPER(freebsd64_shmdt),
 	FREEBSD64_SYSCALL_INIT_HELPER_COMPAT(shmget),
-	FREEBSD64_SYSCALL_INIT_HELPER(freebsd64_shmsys),
 	FREEBSD64_SYSCALL_INIT_HELPER(freebsd64_shmctl),
+#if defined(__i386__) && (defined(COMPAT_FREEBSD4) || defined(COMPAT_43))
```

Comment:
This entry seems odd.  __i386__ will never have freebsd64, so why not just remove this entry without adding the shmsys entry?

Reply:
OTOH, it does seem like the freebsd32_shmsys should be guarded by `#if defined(__amd64__) && ...`?

Reply:
Indeed, it shouldn't be there at all.

I think we could actually drop shmctl support entirely in userspace since it only work for i386 binaries running on amd64 and only with pre-symbol versioning compat support enabled.